rotkehlchen/tasks/manager.py

```diff
@@ -536,8 +546,50 @@ def _maybe_update_snapshot_balances(self) -> Optional[list[gevent.Greenlet]]:
         Update the balances of a user if the difference between last time they were updated
         and the current time exceeds the `balance_save_frequency`.
         """
-        with self.database.conn.read_ctx() as cursor:
-            if self.database.should_save_balances(cursor):
+        with self.database.conn.write_ctx() as write_cursor:
+            if self.database.should_save_balances(write_cursor):
+                if CachedSettings().get_settings().auto_detect_tokens is True:
+                    last_save_time = self.database.get_last_balance_save_time(write_cursor)
+                    detected_tokens = defaultdict(list)
+                    for event_data in write_cursor.execute(
+                        # user addresses that have earliest receive events of distinct assets after last_save_time  # noqa: E501
+                        f'SELECT * FROM  (SELECT MIN(timestamp), {HISTORY_BASE_ENTRY_FIELDS}, '
+                        f'{EVM_EVENT_FIELDS} {EVM_EVENT_JOIN} {filter_ignore_asset_query()} '
+                        'GROUP BY asset) WHERE timestamp >= ?;',
+                        (last_save_time * 1000,),  # to ms
+                    ):
+                        event = EvmEvent.deserialize_from_db(event_data[2:])
+                        if (
+                            event.location not in EVM_EVMLIKE_LOCATIONS or
+                            event.location_label is None or
+                            EVM_ADDRESS_REGEX.fullmatch(event.location_label) is None
+                        ):
+                            log.warning(
```

Comment:
I am not sure I understand the warning here. It's a big if condition and can match for many events right? Doesn't the query above protect you from it? If yes so is this more like an assertion of something that should not happen? I am a but confused here.

Otherwise if this warning is going to be happening often then it's gonna spam the logs.

But from what I see it's more like an assertion?

Reply:
yeah, it's more like an assertion that if we look for evm events and their location label is empty or not a valid evm address, then we log this warning.